opentracing/span.py

```diff
@@ -210,21 +244,166 @@ def __exit__(self, exc_type, exc_val, exc_tb):
                 })
         self.finish()
 
-    def log_event(self, event, payload=None):
-        """DEPRECATED"""
-        if payload is None:
-            return self.log_kv({'event': event})
-        else:
-            return self.log_kv({'event': event, 'payload': payload})
 
-    def log(self, **kwargs):
-        """DEPRECATED"""
-        key_values = {}
-        if 'event' in kwargs:
-            key_values['event'] = kwargs['event']
-        if 'payload' in kwargs:
-            key_values['payload'] = kwargs['payload']
-        timestamp = None
-        if 'timestamp' in kwargs:
-            timestamp = kwargs['timestamp']
-        return self.log_kv(key_values, timestamp)
+class ActiveSpan(BaseSpan):
+    """ActiveSpan represents a OpenTracing span's implementation,
+    wrapping a Span object, layering on in-process propagation capabilities.
+
+    In any given thread there is at most one active span primarily
+    responsible for the work accomplished by the surrounding application code.
+    The ActiveSpan may be accessed via the ActiveSpanSource.active_span
+    property. If the application needs to defer work that should be part of
+    the same Span, the Source provides a ActiveSpan.capture() method that
+    returns an ActiveSpanContinuation; this continuation may be used to
+    re-activate and continue the Span in that other thread.
+
+    ActiveSpan implements a Context Manager API that allows the following
+    usage:
+
+    .. code-block:: python
+
+        with tracer.start_active_span(operation_name='go_fishing') as span:
+            call_some_service()
+
+    In the Context Manager syntax it's not necessary to call span.deactivate()
+    """
+
+    def __init__(self, wrapped):
+        self._wrapped = wrapped
+        self._noop_continuation = ActiveSpanContinuation()
+
+    def deactivate(self):
+        """Mark the end of the active period for the current thread and
+        ActiveSpan. When the last ActiveSpan is deactivated for a given Span,
+        it is automatically finished with a call to Span.finish().
+        """
+        pass
+
+    def capture(self):
+        """Capture a new ActiveSpanContinuation associated with this
+        ActiveSpan and Span. The ActiveSpanContinuation may be used as data
+        in a closure or callback function where the ActiveSpan may be resumed
+        and reactivated.
+
+        IMPORTANT: the caller MUST call ActiveSpanContinuation.activate() and
+        call ActiveSpan.deactivate() or the associated Span will never be
+        finished. That is, calling capture() increments a refcount that must be
+        decremented somewhere else.
+        """
+        return self._noop_continuation
+
+    @property
+    def wrapped(self):
```

Comment:
Correct, this is not in the Java implementation - but I took the route the python's `Span` code takes, by defining properties `tracer` and `context` (`tracer` doesn't exist in the Java's interfaces, for example).

In theory, an `ActiveSpan` should always be wrapping a `Span` object - but I can remove it if you feel this `wrapped` property doesn't belong to the base hierarchy :)